File: Python/String/substring/Adding Spaces to a String.py
class Solution:
    def addSpaces(self, s: str, spaces: List[int]) -> str:
        ans=[]
      #Slicing: Specify the start index and the end index, separated by a colon, to return a part of the string
        ans.append(s[:spaces[0]])
        for i in range(1,len(spaces)):
            ans.append(s[spaces[i-1]:spaces[i]])
        n = len(s)
        ss=len(spaces)
        ans.append(s[spaces[-1]:])
        return ' '.join(ans)
        

File: CPP/String/substring/Adding Spaces to a String.cpp
class Solution {
public:
    string addSpaces(string s, vector<int>& spaces) {
        string ans;
        ans+=s.substr(0,spaces[0]);
        ans+=' ';
        int n =s.size();
        for(int i=1;i<spaces.size();i++){
            ans+=s.substr(spaces[i-1],spaces[i]-spaces[i-1]);
            ans+=' ';
        }
        // Add the final segment after the last space
        int ss= spaces.size();
        ans+=s.substr(spaces[ss-1],n-spaces[ss-1]);
        return ans;
    }
};
